x-pack/plugins/security_solution/common/license/policy_config.ts

```diff
@@ -60,7 +58,11 @@ export const isEndpointPolicyValidForLicense = (
   ) {
     return false;
   }
+  return true;
+}
 
+function isEndpointPolicyValidForRansomware(policy: PolicyConfig): boolean {
```

Comment:
I'm not sure this file's refactor makes sense.

`isEndpointPolicyValidForRansomware(PolicyConfig)` --> By name alone, what does this function do? It reads like it checks if a policy has a valid configuration. What it's actually checking is if modifications were made, from the defaults, which we don't want for certain license levels.

The `isEndpointPolicyValidForLicense` was definitely growing and needed a cleanup. But this locks in some assumptions, and might make it harder if we just allow certain things for Gold, and then more for Plat, etc.

Reply:
@pzl indeed. Perhaps I can instead split each function to accept a policy config and a license and rename the functions to 

`isMalwareEndpointPolicyValidForLicense`
`isRansomwareEndpointPolicyValidForLicense`
etc...

What do you think?

Reply:
@pzl I refactor the function to split them into independent checks for policy and license. The final check results in this:

```typescript
    isEndpointMalwarePolicyValidForLicense(policy, license) &&
    isEndpointRansomwarePolicyValidForLicense(policy, license) &&
    isEndpointMemoryPolicyValidForLicense(policy, license) &&
    isEndpointBehaviorPolicyValidForLicense(policy, license)
```